src/applications/CreateCSVForDICOMs.py

```diff
@@ -62,128 +63,146 @@ def main():
         required=True,
     )
 
-    args = parser.parse_args()
+    return parser.parse_args()
 
-    output_df_for_csv = pd.DataFrame(
-        columns=["SubjectID", "Timepoint", "T1", "T1GD", "T2", "FLAIR"]
-    )
 
-    subject_timepoint_missing_modalities, subject_timepoint_extra_modalities = [], []
+class CSVCreator:
+    def __init__(self, inputDir: str, outputCSV: str):
+        self.inputDir = inputDir
+        self.outputCSV = outputCSV
+        self.subject_timepoint_missing_modalities = []
+        self.subject_timepoint_extra_modalities = []
+        self.output_df_for_csv = pd.DataFrame(
+            columns=["SubjectID", "Timepoint", "T1", "T1GD", "T2", "FLAIR"]
+        )
 
-    for subject in tqdm(os.listdir(args.inputDir)):
-        current_subject_dir = posixpath.join(args.inputDir, subject)
-        if os.path.isdir(current_subject_dir):
-            for timepoint in os.listdir(current_subject_dir):
-                current_subject_timepoint_dir = posixpath.join(
-                    current_subject_dir, timepoint
-                )
-                if os.path.isdir(current_subject_timepoint_dir):
-                    modality_folders = os.listdir(current_subject_timepoint_dir)
-                    # check if there are missing modalities
-                    if len(modality_folders) < 4:
-                        subject_timepoint_missing_modalities.append(
-                            subject + "_" + timepoint
-                        )
-                    # check if there are extra modalities
-                    elif len(modality_folders) > 4:
-                        subject_timepoint_extra_modalities.append(
-                            subject + "_" + timepoint
-                        )
-                    # goldilocks zone
+    def process_data(self):
+        for subject in tqdm(os.listdir(self.inputDir)):
+            self.process_row(subject)
+
+    def process_row(self, subject):
+        inputDir = self.inputDir
```

Comment:
Ensure posix path for all.
```suggestion
        inputDir = posixpath.normpath(self.inputDir)
```